testsupport/wait/host.go

```diff
@@ -2574,3 +2583,60 @@ func (a *HostAwaitility) CreateSpaceAndSpaceBinding(t *testing.T, mur *toolchain
 	})
 	return spaceCreated, spaceBinding, err
 }
+
+// UpdateSocialEvent tries to update the Status or Spec of the given SocialEvent
+// If it fails with an error (for example if the object has been modified) then it retrieves the latest version and tries again
+// Returns the updated SocialEvent
+func (a *HostAwaitility) UpdateSocialEvent(t *testing.T, status bool, socialEventName string, modifySocialEvent func(us *toolchainv1alpha1.SocialEvent)) (*toolchainv1alpha1.SocialEvent, error) {
+	var socialEvent *toolchainv1alpha1.SocialEvent
+	err := wait.PollUntilContextTimeout(context.TODO(), a.RetryInterval, a.Timeout, true, func(ctx context.Context) (done bool, err error) {
+		freshSocialEvent := &toolchainv1alpha1.SocialEvent{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: socialEventName}, freshSocialEvent); err != nil {
+			return true, err
+		}
+
+		modifySocialEvent(freshSocialEvent)
+		if status {
+			// Update the Status
+			if err := a.Client.Status().Update(context.TODO(), freshSocialEvent); err != nil {
+				t.Logf("error updating SocialEvent Status '%s': %s. Will retry again...", socialEventName, err.Error())
+				return false, err
+			}
+		} else {
+			// Update the Spec
+			if err := a.Client.Update(context.TODO(), freshSocialEvent); err != nil {
+				t.Logf("Error updating SocialEvent Spec '%s': %s. Will retry again...", socialEventName, err.Error())
+				return false, err
+			}
+		}
+		socialEvent = freshSocialEvent
+		return true, nil
+	})
+	return socialEvent, err
+}
+
+// NSTemplateTier updates the given NSTemplateTier using the provided modifiers.
+// If it encounters an error (e.g., object has been modified), it retrieves the latest version and retries.
+// Returns the updated NSTemplateTier.
+func (hostAwait *HostAwaitility) UpdateNSTemplateTier(t *testing.T, tierName string, modifyTier func(tier *toolchainv1alpha1.NSTemplateTier)) (*toolchainv1alpha1.NSTemplateTier, error) {
```

Comment:
the receiver names should stay consistent for all methods
```suggestion
func (a *HostAwaitility) UpdateNSTemplateTier(t *testing.T, tierName string, modifyTier func(tier *toolchainv1alpha1.NSTemplateTier)) (*toolchainv1alpha1.NSTemplateTier, error) {
```